lib/rules/utils/ast-utils.js

```diff
@@ -853,14 +853,7 @@ module.exports = {
     isDirectiveComment(node) {
         const comment = node.value.trim();
 
-        return (
-            node.type === "Line" && comment.indexOf("eslint-") === 0 ||
-            node.type === "Block" && (
-                comment.indexOf("global ") === 0 ||
-                comment.indexOf("eslint ") === 0 ||
-                comment.indexOf("eslint-") === 0
-            )
-        );
+        return /^(eslint[- ])|(global )/u.test(comment);
```

Comment:
```suggestion
        return /^(eslint[- ]|global )/u.test(comment);
```

`^` was only in the left alternative.

```js
/^(eslint[- ])|(global )/u.test("this doesn't start with global "); // true
```